base_search_custom_field_filter/models/base.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Tecnativa - Carlos Dauden
+# Copyright 2020 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+from lxml import etree
+
+
+class Base(models.AbstractModel):
+    _inherit = 'base'
+
+    @api.model
+    def _add_custom_filters(self, res, custom_filters):
+        arch = etree.fromstring(res['arch'])
+        for custom_filter in custom_filters:
+            node = False
+            if custom_filter.position_after:
+                node = arch.xpath(
+                    "//field[@name='%s']" % custom_filter.position_after)
+            if not node:
+                node = arch.xpath("//field[last()]")
+            if node:
+                elem = etree.Element('field', {
+                    'name': 'ir_ui_custom_filter_%s' % custom_filter.id,
+                    'string': custom_filter.name,
+                    'filter_domain': (
+                        "[('%s', '=', 'self')]" % custom_filter.expression),
```

Comment:
Remove apostrophes to don't search "self" string literally
```suggestion
                        "[('%s', '=', self)]" % custom_filter.expression),
```

Reply:
Changed here: https://github.com/Tecnativa/server-ux/pull/1

Reply:
Thanks, typo error while transcribing the result of the investigation.